packages/cozy-stack-client/src/SharingCollection.js

```diff
@@ -19,8 +19,48 @@ class SharingCollection extends DocumentCollection {
       data: resp.data.map(normalizeSharing)
     }
   }
+  /**
+   *
+   * Creates a new Sharing. See https://docs.cozy.io/en/cozy-stack/sharing/#post-sharings
+   *
+   * @param {object} params
+   * @param {object} params.document The document to share
+   * @param {string} params.description Description of the sharin
+   * @param {string} params.previewPath The path (route) to see the preview
+   * @param {Array} params.rules The rules defined to the sharing. See https://docs.cozy.io/en/cozy-stack/sharing-design/#description-of-a-sharing
+   * @param {Array} params.recipients Recipients to add to the sharings (will have the same permissions given by the rules defined by the sharing )
+   * @param {Array} params.readOnlyRecipients Recipients to add to the sharings with only read only access
+   * @param {boolean} params.openSharing If someone else than the owner can add a recipient to the sharing
+   */
+  async create({
+    document,
+    description,
+    previewPath,
+    rules,
+    recipients = [],
+    readOnlyRecipients = [],
+    openSharing
+  }) {
+    const resp = await this.stackClient.fetchJSON('POST', '/sharings/', {
+      data: {
+        type: 'io.cozy.sharings',
+        attributes: {
+          description,
+          preview_path: previewPath,
+          open_sharing: openSharing,
+          rules: rules ? rules : getSharingRules(document)
+        },
+        relationships: {
+          recipients: getRecipientsPayload(recipients),
```

Comment:
I think we could have:

```
{
  relationships: {
    recipients: {
      data: recipients.map(toRelationshipItem)
    }
  }
}
```

The `getRecipientsPayload` does not do anything specific for recipients and the `toRelationshipItem` can be useful elsewhere.

Reply:
11e5f8f10d5c8ce780b964cb280488e072b44f58